src/msix/common/AppxManifestObject.cpp

```diff
@@ -383,4 +385,88 @@ namespace MSIX {
         *documentElement = m_dom->GetDocument().As<IMsixElement>().Detach();
         return static_cast<HRESULT>(Error::OK);
     } CATCH_RETURN();
+
+    // Helper to get capabilities from the manifest
+    std::vector<std::string> AppxManifestObject::GetCapabilities(APPX_CAPABILITY_CLASS_TYPE capabilityClass)
+    {
+        // Windows code returns E_INVALIDARG for APPX_CAPABILITY_CLASS_DEFAULT.
+        ThrowErrorIf(Error::InvalidParameter, capabilityClass > APPX_CAPABILITY_CLASS_CUSTOM || capabilityClass < APPX_CAPABILITY_CLASS_GENERAL, 
+            "Invalid capability class.");
+
+        std::vector<std::string> capabilitiesNames;
+        struct _context
+        {
+            APPX_CAPABILITY_CLASS_TYPE capabilityClass;
+            std::vector<std::string>* capabilitiesNames;
+        };
+        _context context = { capabilityClass, &capabilitiesNames};
+
+        // Parse Capability elements.
+        if (capabilityClass != APPX_CAPABILITY_CLASS_CUSTOM)
+        {
+            XmlVisitor visitorCapabilities(static_cast<void*>(&context), [](void* c, const ComPtr<IXmlElement>& capabilitiesNode)->bool
+            {
+                _context* context = reinterpret_cast<_context*>(c);
+                std::string namespaceAlias = capabilitiesNode->GetPrefix();
+                if (namespaceAlias.empty())
+                {
+                    // If no prefix then default namespace for AppxManifest is win10foundation.
+                    namespaceAlias = "win10foundation";
+                }
+                auto name = capabilitiesNode->GetAttributeValue(XmlAttributeName::Name);
+
+                if (context->capabilityClass & APPX_CAPABILITY_CLASS_GENERAL)

```

Comment:
>context->capabilityClass & APPX_CAPABILITY_CLASS_GENERAL [](start = 20, length = 56)

Not treated as flags on Windows.

Reply:
sad life

---
In reply to: [305430864](https://github.com/Microsoft/msix-packaging/pull/179#discussion_r305430864) [](ancestors = 305430864)